pyriemann/embedding.py

```diff
@@ -0,0 +1,134 @@
+"""Embedding covariance matrices via manifold learning techniques."""
+
+import numpy as np
+from sklearn.base import BaseEstimator, TransformerMixin
+from pyriemann.utils.distance import distance
+    
+class Embedding(BaseEstimator, TransformerMixin):
+
+    """Embed SPD matrices into an Euclidean space of smaller dimension.
+
+    It uses diffusion maps to embed the SPD matrices into an Euclidean space.
+    The Euclidean distance between points in this new space approximates 
+    the Diffusion distance (also called commute distance) between vertices 
+    of a graph where each SPD matrix is a vertex. 
```

Comment:
One of my favourite references for this topic is

S.Lafon, Y. Keller, R. Coifman, "Data fusion and multicue data matching by diffusion maps" (2006) on IEEE Transactions of Pattern Analysis and Machine Intelligence.